src/System.Net.Primitives/src/System/Net/CookieContainer.cs

```diff
@@ -1013,24 +1019,29 @@ public int GetCookiesCount()
             int count = 0;
             lock (SyncRoot)
             {
-                foreach (KeyValuePair<string, CookieCollection> pair in m_list)
+                foreach (CookieCollection cc in m_list.Values)
                 {
-                    CookieCollection cc = pair.Value;
                     count += cc.Count;
                 }
             }
             return count;
         }
 
-        public CookieCollection this[string s]
+        public ICollection Values
```

Comment:
Although I see desktop had it (and with no explicit lock) so I guess it is best to add it as you did.